Tests/SQLKitTests/SQLKitTests.swift

```diff
@@ -348,33 +348,59 @@ CREATE TABLE `planets`(`id` BIGINT, `name` TEXT, `diameter` INTEGER, `galaxy_nam
             let foo: Int
             let bar: Double?
             let baz: String
+            let waldoFred: Int?
         }
 
         do {
             let row = TestRow(data: [
                 "id": UUID(),
                 "foo": 42,
                 "bar": Double?.none as Any,
-                "baz": "vapor"
+                "baz": "vapor",
+                "waldoFred": 2015
             ])
 
             let foo = try row.decode(model: Foo.self)
             XCTAssertEqual(foo.foo, 42)
             XCTAssertEqual(foo.bar, nil)
             XCTAssertEqual(foo.baz, "vapor")
+            XCTAssertEqual(foo.waldoFred, 2015)
+        } catch {
+            XCTFail("Could not decode row \(error)")
         }
         do {
             let row = TestRow(data: [
                 "foos_id": UUID(),
                 "foos_foo": 42,
                 "foos_bar": Double?.none as Any,
-                "foos_baz": "vapor"
+                "foos_baz": "vapor",
+                "foos_waldoFred": 2015
             ])
 
-            let foo = try row.decode(model: Foo.self, prefix: "foos_")
+            let foo = try row.decode(model: Foo.self, keyPrefix: "foos_")
             XCTAssertEqual(foo.foo, 42)
             XCTAssertEqual(foo.bar, nil)
             XCTAssertEqual(foo.baz, "vapor")
+            XCTAssertEqual(foo.waldoFred, 2015)
+        } catch {
+            XCTFail("Could not decode row with prefix \(error)")
+        }
+        do {
+            let row = TestRow(data: [
+                "id": UUID(),
+                "foo": 42,
+                "bar": Double?.none as Any,
+                "baz": "vapor",
+                "waldo_fred": 2015
+            ])
+
+            let foo = try row.decode(model: Foo.self, keyDecodingStrategy: .convertFromSnakeCase)
+            XCTAssertEqual(foo.foo, 42)
+            XCTAssertEqual(foo.bar, nil)
+            XCTAssertEqual(foo.baz, "vapor")
+            XCTAssertEqual(foo.waldoFred, 2015)
+        } catch {
+            XCTFail("Could not decode row with keyDecodingStrategy \(error)")
         }
     }
 }
```

Comment:
Would be great to also have a test with the custom key decoding strategy 😄

Reply:
I've added a test for this in the latest commit

Reply:
> Would be great to also have a test with the custom key decoding strategy 😄

I've added a test for this in the latest change.